modules/controller/variables.tf

```
@@ -0,0 +1,171 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+variable "consul_ecs_image" {
+  description = "consul-ecs Docker image."
+  type        = string
+  default     = "ganeshrockz/ecs"
+}
+
+variable "ecs_cluster_arn" {
+  description = "The ARN of the ECS cluster where the controller will be running."
+  type        = string
+}
+
+variable "region" {
+  description = "AWS region of the ECS cluster."
+  type        = string
+}
+
+variable "requires_compatibilities" {
+  description = "Set of launch types required by the task."
+  type        = list(string)
+  default     = ["EC2", "FARGATE"]
+}
+
+variable "launch_type" {
+  description = "Launch type on which to run service. Valid values are EC2 and FARGATE."
+  type        = string
+  default     = "FARGATE"
+}
+
+variable "consul_bootstrap_token_secret_arn" {
+  description = "The ARN of the AWS SecretsManager secret containing the token to be used by this controller. The token needs to have at least `acl:write` and `node:write` privileges in Consul."
+  type        = string
+}
+
+variable "log_configuration" {
+  description = "Task definition log configuration object (https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_LogConfiguration.html)."
+  type        = any
+  default     = null
+}
+
+variable "iam_role_path" {
+  description = "IAM roles at this path will be permitted to login to the Consul AWS IAM auth method configured by this controller."
+  type        = string
+  default     = "/consul-ecs/"
+}
+
+variable "subnets" {
+  description = "Subnets where the controller task should be deployed. If these are private subnets then there must be a NAT gateway for image pulls to work. If these are public subnets then you must also set assign_public_ip for image pulls to work."
+  type        = list(string)
+}
+
+variable "consul_server_address" {
+  description = "Address of the consul server host"
```

This address can also be an `exec=` string. We should update this description to be similar to [consul-k8s](https://developer.hashicorp.com/consul/docs/k8s/helm#v-externalservers-hosts)

```suggestion
  description = "Address of the Consul servers. This can be an IP address, a DNS name, or an `exec=` string specifying a script that outputs IP addresses. Refer to https://github.com/hashicorp/go-netaddrs#summary for details. This variable should not specify the port. Instead, use var.http_config.port and var.grpc_config.port to change the server HTTP and gRPC ports.
```

---

nit: Consul Dataplane calls this setting `-addresses`, and consul-server-connection-manager also uses `Addresses` for the config field. K8s uses `externalServers.hosts` for the same setting (`hosts` was the field name used prior to Consul Dataplane)

IMO, using the plural, "addresses", is slightly confusing because it is not actually a list. To be consistent, should we make this plural as well - `consul_server_addresses`?

---

Oh, I forget we are using `consulServers.hosts` in the ECS config json: https://github.com/hashicorp/consul-ecs/blob/ganeshrockz/agentless/config/schema.json#L85

Then maybe we should match Consul K8s and go with `consul_hosts` or `consul_server_hosts` (I like `hosts` because it indicates the port is not included)

---

`consul_server_hosts` sounds lot more better